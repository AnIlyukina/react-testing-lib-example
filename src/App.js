import {useEffect, useState} from "react";
import AppRouters from "./router/AppRouters";
import Navbar from "./components/Navbar/Navbar";
import Counter from "./components/Counter/Counter";

function App() {
    const [data, setData] = useState(null);
    const [toggle, setToggle] = useState(false);
    const [value, setValue] = useState('')
    const toggleClick = () => setToggle(prev => !prev);

    useEffect(() => {
        setTimeout(() => {
            setData({})
        }, 100)
    }, [])

    return (<div>

        <Navbar/>
        <AppRouters/>
        {/*<p data-testid="value-elem">{ value }</p>*/}
        {/*{toggle && <div data-testid="toggle-div">toggle</div>}*/}
        {/*{data && <div style={{color: 'red'}}>data</div>}*/}
        {/*<h1>Hello world</h1>*/}
        {/*<button data-testid="toggle-btn" onClick={toggleClick}> Кнопка </button>*/}
        {/*<input type="text" placeholder='инпут value' onChange={e => setValue(e.target.value)} />*/}
    </div>);
}

export default App;
