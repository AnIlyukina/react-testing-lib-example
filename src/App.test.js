import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';


describe('TEST  APP', () => {
    test('render btn input', () => {
        render(<App />);
        const helloWorldElement = screen.getByText(/hello world/i);
        const btn = screen.getByRole('button');
        const input = screen.getByPlaceholderText(/инпут/i);
        expect(helloWorldElement).toBeInTheDocument();
        expect(btn).toBeInTheDocument();
        expect(input).toMatchSnapshot();
    });

    // queryByText проверка что такого элемента на странице нет
    test('queryByText', () => {
        render(<App />);
        const notElement = screen.queryByText(/hello12/i);
        expect(notElement).toBeNull();
    });

    // findByText для работы с асинхронным кодом
    test('findByText', async () => {
        render(<App />);
        const dataTextElement = await screen.findByText(/data/i);
        expect(dataTextElement).toBeInTheDocument();
        expect(dataTextElement).toHaveStyle({color: 'red'});
    });

    test('clickEvent', async () => {
        render(<App />);
        const btn = screen.getByTestId('toggle-btn');

        expect(screen.queryByTestId('toggle-div')).toBeNull()
        // искусственное событие
        fireEvent.click(btn)
        expect(screen.queryByTestId('toggle-div')).toBeInTheDocument()

        // искусственное событие
        fireEvent.click(btn)
        expect(screen.queryByTestId('toggle-div')).toBeNull()
    });

    test('inputEvent', async () => {
        render(<App />);
        const input = screen.getByPlaceholderText(/инпут/i);
        expect(screen.queryByTestId('value-elem')).toContainHTML('')

        // искусственное событие
        // fireEvent.input(input, {
        //     target: {
        //         value: '1111'
        //     }
        // })

        // не работает с конкретным событием, а воспроизводит действия пользователей
        userEvent.type(input, '1111')
        expect(screen.queryByTestId('value-elem')).toContainHTML('1111')

    });
})
