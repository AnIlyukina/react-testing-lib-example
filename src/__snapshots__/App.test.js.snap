// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST  APP render btn input 1`] = `
<input
  placeholder="инпут value"
  type="text"
/>
`;

exports[`renders App element render btn input 1`] = `
<input
  placeholder="инпут value"
  type="text"
/>
`;
